### ISSUE TITLE
Unexpected torch.autograd.set_grad_enabled(False) in BayesFlow (dev branch)

### ISSUE THREAD

**Issue by arrjon** [bayesflow-org/bayesflow#355]
I recently loaded bayesflow from the dev branch with the pyTorch backend and spent some time debugging my own code, only to realize that BayesFlow automatically disables gradients when using the pyTorch backend:

```
if keras.backend.backend() == "torch":
    # Turn off gradients by default
    import torch
    torch.autograd.set_grad_enabled(False)
```
This behavior is not explicitly documented or warned about, which can lead to confusion, especially when debugging issues related to autograd.

I suggest that users should be explicitly informed that gradients are disabled by default when using the pyTorch backend.

**Comment by vpratz** [comment]
Fixed by b0af372
